crates/ruff/src/directives.rs

```diff
@@ -95,6 +95,27 @@ fn extract_noqa_line_for(lxr: &[LexResult], locator: &Locator, indexer: &Indexer
                 break;
             }
 
+            // F-strings can be multi-line even without triple quotes. For example,
+            //
+            // ```python
+            // f"foo {
+            //    x
+            //        *
+            //             y
+            // }"
+            // ```
+            //
+            // We expect `noqa` directives on the last line of the f-string.
+            Tok::FStringEnd => {
+                // SAFETY: If we're at the end of a f-string, the indexer must have a
+                // corresponding f-string range for it.
+                let f_string_start = indexer.f_string_range(range.start()).unwrap().start();
```

Comment:
There's a slight performance penalty here as not many single quoted f-strings are going to be multi-line but we can't really know that until we get the entire f-string range.